src/repository/storage/postgres/orm/sequelize/note.ts

```diff
@@ -156,6 +163,20 @@ export default class NoteSequelizeStorage {
     });
   };
 
+  public createAssociationWithNoteRelationModel(model: ModelStatic<NoteRelationsModel>): void {
+    /**
+     * Create association with note relations
+     */
+    this.noteRelationModel = model;
+  }
+
+  public createAssociationWithTeamsModel(model: ModelStatic<TeamsModel>): void {
+    /**
+     * Create association with teams
+     */
+    this.teamModel = model;
+  }
```

Comment:
I'm not sure that you need to access team model here. You can return all note parents by `getNoteParents`. 

If you want to skip notes where user has no access, you can do it on a domain level. But I'm not sure that is is a real case. Is it possible that user has an access to a particular note, but has no access to it's parent?

Reply:
Probably if there was a nested note relation, with different team members, where there could be some missing role access for team member in some specific notes. I might be wrong, that was based on what I understand from the code. 👀

Reply:
I still think, that such a coupling of two independent modules (note and team) is bad idea. Such domains designed to be decoupled from each other. 

My suggestion is to return all note parents. The only check is that user should has and access to the target note — it can be checked on a presentation (http) level via the `notePublicOrUserInTeam` policy

Reply:
it seems like creation of a golem, user would have ability to see notes which he has no access to (we return note with full content)
like in default cases
`
privateNote -> privateNote -> userInThisNoteTeam
`
all `privateNotes` should not be returned in this case but in your suggestion user would have ability to see full note contents of them

Reply:
i think that we can use shared domains and move this logic to domain level as trade-off between speed and architectural purity

Reply:
I think we can keep it simple for the moment, as @neSpecc proposed, by returning all the note parents, when the user try to access the note, and he does not have access to, the return of content will be blocked by the policy used.